diff --git a/Makefile.am b/Makefile.am
index 684d0b3..c3d438c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,28 +16,7 @@ ipl_LDFLAGS = -Wl,--whole-archive,-L.libs,-lipl,-lpdbg,--no-whole-archive
 
 if BUILD_P9
 libipl_istep_SOURCES = \
-	libipl/p9/ipl0.C \
-	libipl/p9/ipl1.C \
-	libipl/p9/ipl2.C \
-	libipl/p9/ipl3.C \
-	libipl/p9/ipl4.C \
-	libipl/p9/ipl5.C \
-	libipl/p9/ipl6.C \
-	libipl/p9/ipl7.C \
-	libipl/p9/ipl8.C \
-	libipl/p9/ipl9.C \
-	libipl/p9/ipl10.C \
-	libipl/p9/ipl11.C \
-	libipl/p9/ipl12.C \
-	libipl/p9/ipl13.C \
-	libipl/p9/ipl14.C \
-	libipl/p9/ipl15.C \
-	libipl/p9/ipl16.C \
-	libipl/p9/ipl17.C \
-	libipl/p9/ipl18.C \
-	libipl/p9/ipl19.C \
-	libipl/p9/ipl20.C \
-	libipl/p9/ipl21.C
+	libipl/p9/ipl0.C
 
 libipl_istep_CXXFLAGS = \
 	-I$(INCDIR)/ekb/chips/p9/procedures/hwp/nest
diff --git a/libipl/p9/ipl0.C b/libipl/p9/ipl0.C
index c896d26..5275258 100644
--- a/libipl/p9/ipl0.C
+++ b/libipl/p9/ipl0.C
@@ -7,15 +7,7 @@ extern "C" {
 #include "libipl.H"
 #include "libipl_internal.H"
 
-#include <ekb/chips/p9/procedures/hwp/perv/p9_pre_poweron.H>
-#include <ekb/chips/centaur/procedures/hwp/perv/cen_pre_poweron.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_select_clock_mux.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_clock_test.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_set_fsi_gp_shadow.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_select_boot_master.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_setup_sbe_config.H>
 #include <ekb/chips/p9/procedures/hwp/perv/p9_start_cbs.H>
-#include <ekb/chips/p9/procedures/hwp/perv/p9_nv_ref_clk_enable.H>
 
 static void ipl_pre0(void)
 {
@@ -29,100 +21,6 @@ static void ipl_pre0(void)
 	}
 }
 
-static int ipl_poweron(void)
-{
-	struct pdbg_target *pib, *memb;
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_pre_poweron(pib);
-	}
-
-	pdbg_for_each_class_target("memb", memb) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		cen_pre_poweron(memb);
-	}
-
-	return 0;
-}
-
-static int ipl_startipl(void)
-{
-	return -1;
-}
-
-static int ipl_set_ref_clock(void)
-{
-	return -1;
-}
-
-static int ipl_proc_clock_test(void)
-{
-	struct pdbg_target *pib;
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_select_clock_mux(pib);
-	}
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_clock_test(pib, BOTH_SRC0);
-	}
-
-	return 0;
-}
-
-static int ipl_proc_prep_ipl(void)
-{
-	struct pdbg_target *pib;
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_set_fsi_gp_shadow(pib);
-	}
-
-	return 0;
-}
-
-static int ipl_proc_select_boot_master(void)
-{
-	struct pdbg_target *pib;
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_select_boot_master(pib);
-	}
-
-	return 0;
-}
-
-static int ipl_sbe_config_update(void)
-{
-	struct pdbg_target *pib;
-
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_setup_sbe_config(pib);
-	}
-
-	return 0;
-}
-
 static int ipl_sbe_start(void)
 {
 	struct pdbg_target *pib;
@@ -134,24 +32,10 @@ static int ipl_sbe_start(void)
 		p9_start_cbs(pib, true);
 	}
 
-	pdbg_for_each_class_target("pib", pib) {
-		if (pdbg_target_status(pib) != PDBG_TARGET_ENABLED)
-			continue;
-
-		p9_nv_ref_clk_enable(pib);
-	}
-
 	return 0;
 }
 
 static struct ipl_step ipl0[] = {
-	{ IPL_DEF(poweron),                 0,  1,  true,  true  },
-	{ IPL_DEF(startipl),                0,  2,  false, true  },
-	{ IPL_DEF(set_ref_clock),           0,  6,  true,  true  },
-	{ IPL_DEF(proc_clock_test),         0,  7,  true,  true  },
-	{ IPL_DEF(proc_prep_ipl),           0,  8,  true,  true  },
-	{ IPL_DEF(proc_select_boot_master), 0, 11,  true,  true  },
-	{ IPL_DEF(sbe_config_update),       0, 13,  true,  true  },
 	{ IPL_DEF(sbe_start),               0, 14,  true,  true  },
 	{ NULL, NULL, -1, -1, false, false },
 };
